### PR TITLE
feat(Google Maps): new presence

### LINKED ISSUE #5372: Google Maps | maps.google.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4567

<div type='discussions-op-text'>

<sup>Originally posted by **MaXfAa** February 26, 2021</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
Maxou#1909

**What's the name of the service?** 
https://www.google.com/maps or https://maps.google.com | Google Maps

**What should the Presence display?** 
1. Google Maps
2. Hover over the area :
3. Brussels, Belgium
or
1. Google Maps
2. Simulates a route
3. From [city] to [city]

**If possible, please provide a logo for the service (512x512 minimum)** 

https://developers.google.com/maps/images/maps-icon.svg</div>

### PR DESCRIPTION
**Pull request by realtofuine**
## Description 

add presence for google maps

closes #5372

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/G/Google Maps/presence.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		clientId: "993496501886136371",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+function lettersOnly(str: string) {
+	return str.replace(/[^a-zA-Z]/g, " ");
+}
+
+presence.on("UpdateData", async () => {
+	// if (document.location.pathname.includes("/maps")) {
```
Comment (EncryptedDev) [inline websites/G/Google Maps/presence.ts:11]: Remove this comment

File: websites/G/Google Maps/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Tofuine",
+		"id": "993496501886136371"
+	},
+	"service": "Google Maps",
+	"description": {
+		"en": "Google Maps is a web mapping platform and consumer application offered by Google. It offers satellite imagery, aerial photography, street maps, 360° interactive panoramic views of streets, real-time traffic conditions, and route planning for traveling by foot, car, bike, air and public transportation."
+	},
+	"url": "www.google.com",
```
Comment (EncryptedDev) [inline websites/G/Google Maps/dist/metadata.json:11]: Would this not interfere with the other Google presences? Sure you can just use `maps.google.com` instead

File: websites/G/Google Maps/presence.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+	};
+	presenceData.startTimestamp = browsingTimestamp;
```
Comment (EncryptedDev) [inline websites/G/Google Maps/presence.ts:15]:
```suggestion
	const presenceData: PresenceData = {
		largeImageKey: "logo",
		startTimestamp: browsingTimestamp
	};
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Add a privacy filter too please

**Comment by realtofuine** [comment]
WAIT DONT REVIEW IT YET I FORGOT PRIVACY MODE

File: websites/G/Google Maps/dist/metadata.json (hunk 2 above)
Comment (EncryptedDev) [reply websites/G/Google Maps/dist/metadata.json:11]: Lol, it does that for me sometimes and on some devices not everything xD

File: websites/G/Google Maps/presence.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+			presenceData.details = `Viewing ${lettersOnly(
+				place.substring(indexes[4] + 1, indexes[5])
+			)}`;
+			presenceData.buttons = [
+				{ label: "View Place", url: document.location.href },
+			];
+		}
+	} else if (document.location.pathname.includes("/dir")) {
+		if (privacy)
+			presenceData.details = "Viewing directions from one place to another";
```
Comment (EncryptedDev) [inline websites/G/Google Maps/presence.ts:33]: Sounds rather conversation
```suggestion
			presenceData.details = "Viewing directions to a location";
```

File: websites/G/Google Maps/presence.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		else {
+			presenceData.details = "Viewing directions";
+			let from, destination;
+			if (document.querySelector("#sb_ifc50 > input") === null) {
```
Comment (EncryptedDev) [inline websites/G/Google Maps/presence.ts:37]:
```suggestion
			if (!document.querySelector("#sb_ifc50 > input")) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]
Looks fine

File: websites/G/Google Maps/presence.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		else {
+			presenceData.details = "Viewing directions";
+			let from, destination;
+			if (!document.querySelector("#sb_ifc50 > input")) {
+				(from = document
+					.querySelector("#sb_ifc51 > input")
+					.getAttribute("aria-label")),
+					(destination = document
+						.querySelector("#sb_ifc52 > input")
+						.getAttribute("aria-label"));
```
Comment (Bas950) [inline websites/G/Google Maps/presence.ts:42]:
```suggestion
				from = document
					.querySelector("#sb_ifc51 > input")
					.getAttribute("aria-label");
					destination = document
						.querySelector("#sb_ifc52 > input")
						.getAttribute("aria-label");
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by EncryptedDev** [comment]
Revert the last two commits; Prettier requires this

**Comment by EncryptedDev** [comment]
@realtofuine Run `yarn format` please

**Comment by realtofuine** [comment]
prettier's love for parentheses is unmatched

**Review by Bas950 — APPROVED** [review]

**Pull request merged**